apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -383,15 +383,23 @@ export const reloadAfterEditingSection = () => (dispatch, getState) => {
   });
 };
 
-export const asyncLoadSectionData = id => dispatch => {
-  dispatch({type: ASYNC_LOAD_BEGIN});
-  // If section id is provided, load students for the current section.
+export const asyncLoadSectionData = (id, withoutStudents) => dispatch => {
   dispatch({type: ASYNC_LOAD_BEGIN});
+
+  let sectionsSummaryApi = '/dashboardapi/sections';
+  // When loading section summary on teacher homepage,
+  // do not include student-level data.
+  if (withoutStudents) {
+    sectionsSummaryApi += '?without_students=true';
+  }
+
   let apis = [
-    '/dashboardapi/sections',
+    sectionsSummaryApi,
     '/dashboardapi/sections/valid_course_offerings',
     '/dashboardapi/sections/available_participant_types'
   ];
+
+  // If section id is provided, load students for the current section.
```

Comment:
Oh hmmm.  It seems like there is a relationship between `id` and `without_students`.  Maybe we should check `without_students` on the next line?

Oh, on second thought...  Does this next line mean that the students data returned by sections is never used?  (If it is being used, why are we getting the list of students twice?)

Reply:
> Oh hmmm. It seems like there is a relationship between id and without_students. Maybe we should check without_students on the next line?

Do you mean that the variable naming/comments imply that there's a relationship between them? Or that there is some functional relationship between them? 

> Oh, on second thought... Does this next line mean that the students data returned by sections is never used? (If it is being used, why are we getting the list of students twice?)

I'm not sure if all calls to `asyncLoadSectionData` do not use the student-level data returned by `/dashboardapi/sections`. I basically just tried to write this in a way that I was confident would not affect other usages, without actually looking closely at them :).

Reply:
Could rename parameter to `sectionsSummaryWithoutStudents` if that's more clear?